src/transformers/models/llava/modeling_llava.py

```diff
@@ -305,17 +305,22 @@ def get_image_features(
             raise ValueError(f"Unexpected select feature strategy: {self.config.vision_feature_select_strategy}")
 
         kwargs = {k: v for k, v in kwargs.items() if v is not None}
-        # this is not memory efficient at all (output_hidden_states=True) will save all the hidden states.
-        image_outputs = self.vision_tower(pixel_values, output_hidden_states=True, **kwargs)
+        image_outputs = self.vision_tower(
+            pixel_values,
+            output_hidden_states=[vision_feature_layer]
+            if isinstance(vision_feature_layer, int)
+            else vision_feature_layer,
+            **kwargs,
+        )
```

Comment:
We are trying to keep var definition outside of function call

```suggestion
        output_hidden_states = [vision_feature_layer] if isinstance(vision_feature_layer, int) else vision_feature_layer
        image_outputs = self.vision_tower(pixel_values, output_hidden_states=True, **kwargs)
```